### PR TITLE
enh(conf/export) export host & service severities

### PR DESCRIPTION
**Pull request by a-launois**
## Description

export host & services severities  in new export file severities.cfg

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

<h2> How this pull request can be tested ? </h2>

- create severities linked to host/services and enabled
- export configuration
- new file should be present with other export files

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/class/config-generate/severity.class.php [hunk 1]
```diff
@@ -276,4 +297,34 @@ public function getServiceSeverityMappingHostSeverityByName($hc_name)
         $this->service_severity_cache[$hc_name] = &$severity;
         return $severity['sc_id'];
     }
+
+    /**
+     * Export cached objects in corresponding export file
+     */
+    public function generateObjects(): void
+    {
+        foreach ($this->service_severity_cache as $id => $value) {
```
Comment (kduret) [inline www/class/config-generate/severity.class.php:306]: you should split in 2 private functions : `generateServiceSeverities` & `generateHostSeverities`

File: www/class/config-generate/severity.class.php [hunk 2]
```diff
@@ -46,11 +46,30 @@ class Severity extends AbstractObject
...
+        'level' => 'level',
+        'icon_id' => 'icon_id',
+    ];
+    protected $attributes_write = array(
+        'id',
+        'name',
+        'level',
+        'icon_id',
+        'type'
+    );
```
Comment (kduret) [inline www/class/config-generate/severity.class.php:72]:
```suggestion
    protected $attributes_write = [
        'id',
        'name',
        'level',
        'icon_id',
        'type',
    ];
```

File: www/class/config-generate/severity.class.php [hunk 3]
```diff
@@ -276,4 +297,50 @@ public function getServiceSeverityMappingHostSeverityByName($hc_name)
...
+    /**
+     * Export service severities in corresponding export file
+     */
+    private function generateServiceSeverityObjects(): void
+    {
+        foreach ($this->service_severity_cache as $id => $value) {
+            if (is_null($value) || ! in_array($id, $this->service_linked_cache)) {
+                continue;
+            }
+            $severity = [];
```
Comment (kduret) [inline www/class/config-generate/severity.class.php:320]:
```suggestion
            $severity = ['type' => 'service'];
```

File: www/class/config-generate/severity.class.php [hunk 4]
```diff
@@ -276,4 +297,50 @@ public function getServiceSeverityMappingHostSeverityByName($hc_name)
...
+            foreach ($this->attributesSelectService as $selectAttr => $writeAttr) {
+                $severity[$writeAttr] = $value[$selectAttr];
+            }
+            $severity['type'] = 'service';
```
Comment (kduret) [inline www/class/config-generate/severity.class.php:324]:
```suggestion

```

File: www/class/config-generate/severity.class.php [hunk 5]
```diff
@@ -276,4 +297,50 @@ public function getServiceSeverityMappingHostSeverityByName($hc_name)
...
+    /**
+     * Export host severities in corresponding export file
+     */
+    private function  generateHostSeverityObjects(): void
+    {
+        foreach ($this->host_severity_cache as $id => $value) {
+            if (is_null($value) || ! in_array($id, $this->host_linked_cache)) {
+                continue;
+            }
+            $severity = [];
```
Comment (kduret) [inline www/class/config-generate/severity.class.php:338]:
```suggestion
            $severity = ['type' => 'host'];
```

File: www/class/config-generate/severity.class.php [hunk 6]
```diff
@@ -276,4 +297,50 @@ public function getServiceSeverityMappingHostSeverityByName($hc_name)
...
+            foreach ($this->attributesSelectHost as $selectAttr => $writeAttr) {
+                $severity[$writeAttr] = $value[$selectAttr];
+            }
+            $severity['type'] = 'host';
```
Comment (kduret) [inline www/class/config-generate/severity.class.php:342]:
```suggestion

```

File: www/class/config-generate/severity.class.php [hunk 7]
```diff
@@ -276,4 +297,50 @@ public function getServiceSeverityMappingHostSeverityByName($hc_name)
...
+            }
+            $severity['type'] = 'service';
+            $this->generateObjectInFile($severity, $id);
+        }
+    }
+
+    /**
+     * Export host severities in corresponding export file
+     */
+    private function  generateHostSeverityObjects(): void
```
Comment (kduret) [inline www/class/config-generate/severity.class.php:332]:
```suggestion
    private function generateHostSeverityObjects(): void
```

**Review by kduret — APPROVED** [review]

**Review by callapa — APPROVED** [review]

**Review by kduret — APPROVED** [review]

**Pull request merged**